pkgs/development/compilers/gcc/common/pre-configure.nix

```diff
@@ -54,6 +54,22 @@ lib.optionalString (hostPlatform.isSunOS && hostPlatform.is64bit) ''
   export STRIP='strip -x'
 ''
 
+# We need to reconfigure the build system for mlibc targets, mostly because
+# libstdcxx tries running link tests (which we can't do without a libc when
+# we're bootstrapping mlibc). This should probably only be required when
+# withoutTargetLibc, however managarm's recipes do this for every gcc build.
+# This should probably also be done for other targets when withoutTargetLibc,
+# but no other target seems to have issues without doing this so might as well
+# leave them untouched.
```

Comment:
Generally it's much better to avoid platform conditionals where we can, as it means fewer divergences in configuration that could be overlooked when making a change.

Reply:
I always say, a mass rebuild only happens once, but code quality is forever. With exceptions, it's not a good idea to accept worse code forever to avoid a single mass rebuild.

Reply:
Ok then I'll be retargeting this PR to staging tomorrow. I'm also not very sure if this would cause issues with other targets when withoutTargetLibc, but I guess there's no other way to know